### PR TITLE
Cookies and Sessions in ASP.NET

### LINKED ISSUE #5879: Cookies and Sessions in ASP dot NET
### Topic Suggestion
Cookies and Sessions in ASP dot NET

### Proposed title of article
Cookies and Sessions in ASP dot NET

### Proposed article introduction
Cookies are small tokens of data that is stored on a client machine. Are utilized to store user data on the client end. Name, user id, contacts, address etc are samples of data stored on the client machine. ASP dot NET provides two types of cookies, Non-persistent and persistent. Cookies that have expiry date are the referred to as persistent, those that don't expire are known as Non-persistent.

Session is a state from where user values are retrieved and kept in the webpage. It is used to pass data from one page to another in ASP dot NET.
In a website sessions allows user to interact with the application and perform operations such as read and write.

### Key takeaways
- Creating cookies in ASP dot NET.
- Properties of cookies.
- Initiation and termination of a session.
- Implementation of sessions in ASP dot NET.

### Article quality
In this article I will introduce user to sessions and cookies in ASP.NET, am going to take the user through creation of cookies, also am going to show the reader how to implement a session in ASP dot NET. Further more am also going to cover properties of cookies, how sessions start and end and what leads to them to start and end. Am hoping this article will be of great value to the engineering community.

### References
N/A

### PR DESCRIPTION
**Pull request by swambui1**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #5879

### CODE REVIEW

**Comment by tesswambui** [comment]
@WanjaMIKE  can you have a look at this PR, will appreciate. has been over a week and no one has been assigned this PR.

**Comment by swambui1** [comment]
@hectorkambow  @ahmadmardeni1 , this pr has been here over a week and has not been assigned for review, kindly help.

**Comment by LinusMuema** [comment]
plagiarism check done ✅

File: content/articles/cookies-and-sessions-in-asp-dot-net/index.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+Below are some common properties of cookies in ASP.NET:
+
+- Value, manipulates a single cookie. 
+- values, handles the collection of cookies and manipulates a collection of cookies.
+- Expire, used to set the expiry date of cookies.
+- Domain, associates cookies to domain.
+
+### Initiation and termination of a session
+Session gets initiated when a user hits an application for the first time.
```
Comment (Jethro-magaji) [inline content/articles/cookies-and-sessions-in-asp-dot-net/index.md:113]: A session gets ........

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi! @swambui1 
Well done for such an amazing article. However, there are some corrections you need to make.
 Run your article on tools like Grammarly and ProWritingAid for better editing. This would save the time to review your work.
Here's a [link](https://github.com/section-engineering-education/engineering-education/pull/6210/commits/acc44893d4ec258f4e077b9ce8e3e2ca2b9d6a9f) to a PR with little or no revision to guide you on formatting. However, if you have any questions or need help, you can reach out to me on the EngEd Slack channel.
Thanks.

File: content/articles/cookies-and-sessions-in-asp-dot-net/index.md (hunk 1 above)
Comment (swambui1) [reply content/articles/cookies-and-sessions-in-asp-dot-net/index.md:113]: @Jethro-magaji on it right away

**Comment by swambui1** [comment]
@Jethro-magaji  am done with the requested changes, you can proceed with the review thank you.

**Comment by hectorkambow** [comment]
@swambui1 
Please be sure to proofread the entire article as Peer Reviewers do not provide revisions services (and may not point out all pending issues).
If any other edits are required @Jethro-magaji we can close the PR to allow the student the proper time to make the edits.

**Comment by swambui1** [comment]
@hectorkambow, my reviewer is currently working on this pr

**Comment by hectorkambow** [comment]
cc @Jethro-magaji

**Comment by Jethro-magaji** [comment]
> cc @Jethro-magaji 

Am currently working on the PR.

**Review by Jethro-magaji — APPROVED** [review]
Hi! @swambui1 
Well done for coming up with such an amazing article. Please take note of these corrections to minimize revision next time.
Thanks.

**Comment by WanjaMIKE** [comment]
Merged #6807 into master

**Pull request merged**